src/components/toolbar/AssignVeileder/TildelVeileder.tsx

```diff
@@ -105,36 +101,43 @@ const TildelVeileder = (props: ToolbarProps) => {
     });
   };
 
+  const onBlur = (e: any) => {
+    const currentTarget = e.currentTarget;
+    setTimeout(() => {
+      if (!currentTarget.contains(document.activeElement)) {
+        setState({
+          ...state,
+          chosenVeilederIdent: '',
+          input: '',
+          showList: false,
+          veilederIsChosen: false,
+        });
+      }
+      }, 0);
+  };
+
   const lowerCaseInput = state.input.toLowerCase();
-  const veiledereWithoutCurrentVeileder = removeCurrentVeilederFromVeiledere(props.veiledere, props.aktivVeilederInfo.ident);
-  const veiledereSortedAlphabetically = sortVeiledereAlphabeticallyBySurname(veiledereWithoutCurrentVeileder);
+  const veiledereSortedAlphabetically = sortVeiledereAlphabeticallyWithGivenVeilederFirst(props.veiledere, props.aktivVeilederInfo.ident);
   const filteredVeiledere = filterVeiledereOnInput(veiledereSortedAlphabetically, lowerCaseInput);
 
-  return (<>
-    <FlatButton onClick={() => setState({ ...state, showList: !state.showList })}
-                disabled={hasNoCheckedPersoner(props.markertePersoner)}>{texts.assignOtherVeileder}</FlatButton>
-    {state.showList && <ButtonPanel>
-      <Input label="" value={state.input} type="text" onChange={(event) => onChangeHandler(event)} />
-      <RadioPanelGroup>
-        <VeilederRadioButtons
-          onChangeHandler={radiobuttonOnChangeHandler}
-          filteredVeiledere={filteredVeiledere}
-        />
-      </RadioPanelGroup>
-      <ConfirmVeilederButton
-        chosenVeilederIdent={state.chosenVeilederIdent}
-        confirmVeilederButtonHandler={confirmVeilederButtonHandler}
-        veilederIsChosen={state.veilederIsChosen} />
-    </ButtonPanel>}
-
-    {props.tabType === OverviewTabType.ENHET_OVERVIEW && (
-      <AssignToCurrentVeilederButton
-        aktivVeilederIdent={props.aktivVeilederInfo.ident}
-        confirmVeilederButtonHandler={confirmVeilederButtonHandler}
-        disabled={hasNoCheckedPersoner(props.markertePersoner)}
+  return (<div tabIndex={1} onBlur={onBlur}>
+    <OpenDropdownButton
+      onClick={assignToOtherVeilederButtonHandler}
+      showList={state.showList}
+      userIsChecked={!hasNoCheckedPersoner(props.markertePersoner)}
     />
-    )}
-  </>);
+
+    {state.showList && <Dropdown
+      chooseButtonHandler={chooseButtonHandler}
+      chosenVeilederIdent={state.chosenVeilederIdent}
+      closeButtonHandler={closeButtonHandler}
```

Comment:
Mulig å sortere disse slik at det blir litt lettere å lese?

Reply:
Det kan man sikkert, men nå er de jo alfabetisk da, skal jeg bare legge de i "tilfeldig" rekkefølge, så de ikke kommer rett etter hverandre? Eller er dette et symptom på for dårlig navngivning?

Reply:
Jeg vet ikke, merket bare at det ble litt mye for øynene mine...

Reply:
https://jaketrent.com/post/naming-event-handlers-react/